src/app/kademlia-haskell/release2.nix

```diff
@@ -8,7 +8,13 @@
 
 # Library profiling is on by default, and approximately doubles compile time.
 # Turn it off.
-((import ./packages.nix {}).override
+let pinnedNixpkgs = import (builtins.fetchTarball {
+  name = "nixpkgs-unstable-2019-03-18";
+  url = https://github.com/nixos/nixpkgs/archive/0125544e2a0552590c87dca1583768b49ba911c0.tar.gz;
+  sha256 = "04xvlqw3zbq91zkfa506b2k1ajmj7pqh3nvdh9maabw6m5jhm5rl";
+  });
```

Comment:
We'll lose the unstable cache if we do this right? Does the s3 caching you set up mitigate that somewhat? If so, this is 👍

Reply:
Yeah adding the s3 cache means it will always be cached if we've built the Haskell Kademlia executable on CircleCI before. I uploaded the Linux outputs to the cache too, but that relies on me remembering to do it if things change.